source/common/rds/rds_route_config_subscription.h

```diff
@@ -54,7 +54,7 @@ class RdsRouteConfigSubscription : Envoy::Config::SubscriptionCallbacks,
 
   ~RdsRouteConfigSubscription() override;
 
-  absl::optional<RouteConfigProvider*>& routeConfigProvider();
+  RouteConfigProvider*& routeConfigProvider() { return route_config_provider_; }
```

Comment:
Is the expectation to use this to change the value of `route_config_provider_`? Seems confusing to me to return a reference if that is not the case.

Reply:
Yes, it is used to update the value. The same way the optional value was updated before (via a reference). I did contemplate adding a setter instead, but opted for consistency in this class -- if you look at other member methods they are also references.